independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/IndexClassLookupUtils.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.arc.processor;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.logging.Logger;
+
+public class IndexClassLookupUtils {
+
+    private static final Logger LOGGER = Logger.getLogger(IndexClassLookupUtils.class);
+
+    private IndexClassLookupUtils() {
+    }
+
+    public static ClassInfo getClassByName(IndexView index, DotName dotName) {
+        return lookupClassInIndex(index, dotName, true);
+    }
+
+    /**
+     * Used by {@code BeanArchives.IndexWrapper#getClassByName()} while gathering additional classes for indexing
+     */
+    public static ClassInfo getClassByNameNoLogging(IndexView index, DotName dotName) {
+        return lookupClassInIndex(index, dotName, false);
+    }
+
+    private static ClassInfo lookupClassInIndex(IndexView index, DotName dotName, boolean withLogging) {
+        if (dotName == null) {
+            throw new IllegalArgumentException("Cannot lookup class, provided DotName was null.");
+        }
+        if (index == null) {
+            throw new IllegalArgumentException("Cannot lookup class, provided Jandex Index was null.");
+        }
+        ClassInfo info = index.getClassByName(dotName);
+        if (info == null && withLogging) {
+            // class not in index, log warning as this may cause the application to blow up or behave weirdly
+            LOGGER.info("Class for name: " + dotName + " was not found in Jandex index. Please ensure the class " +
```

Comment:
Shouldn't we use the WARN level instead?

Reply:
That's what I had there originally, but since some pieces of code just choose to "silently" proceed now, WARN felt a bit too strong. But  I wouldn't mind...

Reply:
Hm, that's a tricky question. Let's keep INFO for now...

Reply:
I mean, even `INFO` was noticeable enough for me to fix all those `arc/processor` tests ;)

Reply:
I mean, even INFO made me notice and fix those `arc/processor` tests which were having unindexed classes ;-)